upstairs/src/lib.rs

```diff
@@ -3301,7 +3305,7 @@ impl Downstairs {
                 self.io_state_count.incr(&IOState::Skipped, client_id);
                 self.ds_skipped_jobs[client_id as usize].insert(*ds_id);
```

Comment:
This initially surprised me: it's capturing `&mut self`, which is `Downstairs` here, right?

Reply:
Oh, maybe you're confused by how we can call a `&mut self` function on `self.ds_active`, while also mutating other members of `self`?

That's allowed by the Rust 2021 [disjoint capture](https://doc.rust-lang.org/edition-guide/rust-2021/disjoint-capture-in-closures.html) behavior.

Reply:
My confusion stems from seeing `self` in a closure :)